tests/tornado/test_connection.py

```diff
@@ -81,3 +82,49 @@ def test_close_callback_is_called():
     conn.close()
 
     assert (yield cb_future)
+
+
+@pytest.mark.gen_test
+def test_pending_outgoing():
+    server = TChannel('server')
+    server.listen()
+
+    @server.raw.register
+    def hello(request):
+        assert server._dep_tchannel.peers.peers[0].total_outbound_pendings == 1
+        return 'hi'
+
+    client = TChannel('client')
+    yield client.raw(
+        hostport=server.hostport,
+        body='work',
+        endpoint='hello',
+        service='server'
+    )
+
+    client_peer = client._dep_tchannel.peers.peers[0]
+    server_peer = server._dep_tchannel.peers.peers[0]
+    assert client_peer.total_outbound_pendings == 0
+    assert server_peer.total_outbound_pendings == 0
+
+    class FakeMessageFactory(MessageFactory):
+        def build_raw_message(self, context, args, is_completed=True):
+            assert client_peer.total_outbound_pendings == 1
+            return super(FakeMessageFactory, self).build_raw_message(
+                context, args, is_completed,
+            )
+
+    client_conn = client_peer.connections[0]
```

Comment:
`peer.connect()` will get the connection off the peer. You don't need to access `connections` (we should really hide that behind a `_` anyway).